src/test/html/html-document_test.ts

```diff
@@ -17,105 +17,501 @@ import * as dom5 from 'dom5';
 import * as fs from 'fs';
 import * as parse5 from 'parse5';
 import * as path from 'path';
-import {HtmlParser} from '../../html/html-parser';
+
+import {Analyzer} from '../../analyzer';
 import {ParsedHtmlDocument} from '../../html/html-document';
+import {HtmlParser} from '../../html/html-parser';
+import {SourceRange} from '../../model/model';
+import {FSUrlLoader} from '../../url-loader/fs-url-loader';
+import {WarningPrinter} from '../../warning/warning-printer';
 
 suite('ParsedHtmlDocument', () => {
+  const parser: HtmlParser = new HtmlParser();
+  const url = './source-ranges/html-complicated.html';
+  const basedir = path.join(__dirname, '../static/');
+  const file = fs.readFileSync(path.join(basedir, `${url}`), 'utf8');
+  const document: ParsedHtmlDocument = parser.parse(file, url);
+  const analyzer = new Analyzer({urlLoader: new FSUrlLoader(basedir)});
+  const warningPrinter = new WarningPrinter(null as any, {analyzer});
 
-  suite('sourceRangeForNode()', () => {
-    const parser: HtmlParser = new HtmlParser();
-    const url = '/static/source-ranges/html-complicated.html';
-    const file = fs.readFileSync(
-        path.resolve(__dirname, `../${url}`), 'utf8');
-    const document: ParsedHtmlDocument = parser.parse(file, url);
+  const getUnderlinedText = async(range: SourceRange | undefined) => {
+    if (range == null) {
```

Comment:
This could throw an error, but IMO the error message is nicer by returning this string rather than throwing. 

e.g. this error message

```diff
 - expected, + actual
+  No source range produced.
-     <!-- Single Line Comment -->
-     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

That's easier to track down where the problem is than a message that mentions the error and a line number.